lib/reversetunnel/srv.go

```diff
@@ -576,10 +576,41 @@ func (s *server) Close() error {
 }
 
 func (s *server) Shutdown(ctx context.Context) error {
-	err := s.srv.Shutdown(ctx)
+	var errs []error
+
+	// Ensure listener is closed before sending reconnects.
+	err := s.srv.Close()
+	if err != nil {
```

Comment:
```suggestion
	if err != nil && !utils.IsUseOfClosedNetworkError(err) {
```

Reply:
Changed this around a bit so I don't think we will need this check.